fs.go

```diff
@@ -0,0 +1,161 @@
+package simpleblob
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+// fsInterfaceWrapper wraps an Interface and implements fs.FS.
+type fsInterfaceWrapper struct{ Interface }
+
+// fsBlobWrapper represents data upstream and implements both fs.File
+// and fs.FileInfo for convenience.
+type fsBlobWrapper struct {
+	b      *Blob
+	parent *fsInterfaceWrapper
+	r      *bytes.Reader
+}
+
+// AsFS casts the provided interface to a fs.FS interface if supported,
+// else it wraps it to replicate its functionalities.
+func AsFS(st Interface) fs.FS {
+	if fsys, ok := st.(fs.FS); ok {
+		return fsys
+	}
+	return &fsInterfaceWrapper{st}
+}
+
+// find gets a Blob in wrapped Interface and returns it wrapped as a fsBlobWrapper.
+func (stw *fsInterfaceWrapper) find(name string) (*fsBlobWrapper, error) {
+	blobs, err := stw.Interface.List(context.TODO(), name)
+	if err != nil {
+		return nil, err
+	}
+	for _, b := range blobs {
+		if b.Name == name {
+			return &fsBlobWrapper{&b, stw, nil}, nil
+		}
+	}
+	return nil, os.ErrNotExist
+}
+
+// Open retrieves a Blob, wrapped as a fs.File, from the underlying Interface.
+func (stw *fsInterfaceWrapper) Open(name string) (fs.File, error) {
+	if ret, err := stw.find(name); err != nil {
+		return nil, err
+	} else if ret != nil {
+		return ret, nil
+	}
+	return nil, &fs.PathError{
+		Op:   "open",
+		Path: name,
+		Err:  os.ErrNotExist,
+	}
+}
+
+// ReadFile implements fs.ReadFileFS on top of an Interface wrapped as a fs.FS.
+func (stw *fsInterfaceWrapper) ReadFile(name string) ([]byte, error) {
+	return stw.Load(context.Background(), name)
+}
+
+// Stat implements fs.StatFS on top of an Interface wrapped as a fs.FS.
+func (stw *fsInterfaceWrapper) Stat(name string) (fs.FileInfo, error) {
+	if ret, err := stw.find(name); err != nil {
+		return nil, err
+	} else if ret != nil {
+		return ret.Stat()
+	}
+	return nil, nil
+}
+
+func checkDirPath(name, op string) *fs.PathError {
+	if name != "" && name != "." {
+		return &fs.PathError{Op: op, Path: name, Err: errors.New("no subdirectory available")}
+	}
+	return nil
+}
+
+// ReadDir satisfies fs.ReadDirFS
```

Comment:
It might be best if we do not implement extra FS interface and instead stick to the basic `fs.FS`, but ReadDirFS could be a useful addition and corresponds to the `List` call.

I'm not sure if the single root directory restriction is problematic or not, I can imagine that we may want to store blocs with `/` in the future. Currently we do not clearly describe what kind of names are allowed and this could vary by backend, which is probably not a good thing.